### PR TITLE
feat: add otel support for ExtractSampledTraceID

### PR DESCRIPTION
**Pull request by jpkrohling**
I did a pair-programming session this morning with @leveyjam to sort out a problem with trace IDs not being showed up on Faro logs, and we ended up with this set of changes for mimir-proxies. 

There are two things to work out:

1) the order of the tests matter, and they shouldn't -- we couldn't find an easy way to reset the tracer provider to a situation where it returns a noop span after a new tracer provider was properly initialized (as it's the case for the second/third OTel tests)
2) use this branch as a dependency on Faro to ensure that the code actually solves their problem

### CODE REVIEW

**Comment by leveyjam** [comment]
Thanks @jpkrohling ! I will fix those minor items 
(note if anyone else wants this sooner: `git fetch origin pull/28/head:<YOUR_NEW_BRANCHNAME>` gets you the changes from this PR - see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally )

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/grafana/mimir-proxies
 
-go 1.20
+go 1.19
```
Comment (leveyjam) [inline go.mod:3]:
```suggestion
go 1.20
```
Oh yeah - this'll need to go 🤣

**Review by leveyjam — COMMENTED** [review]
LGTM - will work on test improvements before requesting review (should we convert to draft?)

File: pkg/server/middleware/tracer_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	return ctx, func() {
+		sp.End()
+		otel.SetTracerProvider(previous)
+	}
+}
+
+func getContextWithOpenTelemetryNoop(t *testing.T) (context.Context, func()) {
+	ctx, sp := otel.Tracer("test").Start(context.Background(), "test")
+
+	// sanity check
```
Comment (leveyjam) [inline pkg/server/middleware/tracer_test.go:118]:
```suggestion
	// sanity check IsValid() return value
```
(making a suggestion to kick off the CI again - there are now rules on the PR title)

**Comment by leveyjam** [comment]
🎉 - FYI - there are new linting rules in this repo - PR titles must start with "fix: chore: feat:", etc, and be lower case. (or at least not start with a capital letter after the classifier).

File: pkg/server/middleware/tracer.go [hunk 3]
```diff
@@ -43,29 +44,29 @@ func (t Tracer) Wrap(next http.Handler) http.Handler {
...
-		return "", false
-	}
-	sctx, ok := sp.Context().(jaeger.SpanContext)
-	if !ok {
-		return "", false
-	}
-
-	return sctx.TraceID().String(), true
+	traceID, _ := ExtractSampledTraceID(ctx)
+	return traceID, true
```
Comment (mar4uk) [inline pkg/server/middleware/tracer.go:48]: the logic of `ExtractTraceID` has changed. Now it always returns true in the second parameter. But it probably shouldn't when ExtractSampledTraceID returns `"", false`

Comment (leveyjam) [reply pkg/server/middleware/tracer.go:48]: (That was actually part of the original code - I'll remove the method entirely, it just returned "true" as the "ok" return value (not sampled)).

Comment (leveyjam) [reply pkg/server/middleware/tracer.go:48]: No I won't, in fact... It's actually documented as doing that...
```
// ExtractSampledTraceID works like ExtractTraceID but the returned bool is only
// true if the returned trace id is sampled.
```
=> the return value for the other method is "true" (always)
Ignore my previous comment

**Comment by leveyjam** [comment]
Closing in favor of https://github.com/grafana/mimir-proxies/pull/46

**Pull request closed without merging**